src/datasources/config-api/config-api.service.ts

```diff
@@ -13,20 +16,37 @@ export class ConfigApi {
     private readonly dataSource: CacheFirstDataSource,
     @Inject(IConfigurationService)
     private readonly configurationService: IConfigurationService,
+    private readonly validationErrorFactory: ValidationErrorFactory,
   ) {
     this.baseUri =
       this.configurationService.getOrThrow<string>('safeConfig.baseUri');
   }
 
   async getChains(): Promise<Page<Chain>> {
     const key = 'chains'; // TODO key is not final
-    const url = this.baseUri + '/api/v1/chains';
-    return await this.dataSource.get(key, url);
+    const url = `${this.baseUri}/api/v1/chains`;
+    const page: Page<Chain> = await this.dataSource.get(key, url);
+
+    if (!page?.results.every((chain) => isValidChain(chain))) {
+      // TODO: probably we want to invalidate cache at this point
+      const errors = isValidChain.errors as DefinedError[];
```

Comment:
Here `errors` seems to be a property of `isValidChain` (which is global).

Given a scenario where:
1. First call `isValidChain(chain_error)` should return error
2. Second call `isValidChain(chain_success)` should be successful.

Would the second call also return an error? Ie.: how are the `errors` state managed?

Reply:
As I commented below, `every` exits immediately when the condition is met, so `errors` should be pointed to the failed item.

Reply:
With this I actually meant calling `isValidChain` in different contexts. Even if the first "call"/"context" results in an error.

Asking because this seems to be a "lone" `const` global reference and I'm not sure how the `errors` state is cleared.

Reply:
I don't like a lot how AJV manages the errors in this case, but it attaches an `errors` property to the validation function. This `errors` is an array containing the error(s) for the item which didn't pass the validation. Since `every()` is retuning early it is guaranteed that the loop will exit and `errors` property will contain the errors for the failed item. `const errors = ...` will be called just one time, and will be freed after the function throws, I don't see any possible inconsistent state here... 🤔

Reply:
More here: https://ajv.js.org/guide/getting-started.html

<img width="762" alt="Screenshot 2022-09-02 at 11 45 54" src="https://user-images.githubusercontent.com/21182069/188116380-e0feb9fd-6060-4312-9733-e01ff48c10d2.png">

Reply:
Awesome! That was it. So it holds the `errors` state until the next `validate` call! That was my question yeah! 👍